astronomer/providers/amazon/aws/operators/sagemaker.py

```diff
@@ -71,137 +59,19 @@ def __init__(self, **kwargs: Any) -> None:
-            )
-
-            self.log.info(secondary_training_status_message(description, None))
-
-            if status in self.hook.failed_states:
-                reason = last_description.get("FailureReason", "(No reason provided)")
-                raise AirflowException(f"SageMaker job failed because {reason}")
-            elif status == "Completed":
-                billable_time = (
-                    last_description["TrainingEndTime"] - last_description["TrainingStartTime"]
-                ) * instance_count
-                self.log.info(
-                    f"Billable seconds: {int(billable_time.total_seconds()) + 1}\n"
-                    f"{self.task_id} completed successfully."
-                )
-                return {"Training": serialize(description)}
-
-            self.defer(
-                timeout=self.execution_timeout,
-                trigger=SagemakerTrainingWithLogTrigger(
-                    poke_interval=self.check_interval,
-                    end_time=end_time,
-                    aws_conn_id=self.aws_conn_id,
-                    job_name=self.config["TrainingJobName"],
-                    instance_count=int(instance_count),
-                    status=status,
-                ),
-                method_name="execute_complete",
-            )
-        else:
-            if status in self.hook.failed_states:
-                raise AirflowException(f"SageMaker job failed because {description['FailureReason']}")
-            elif status == "Completed":
-                self.log.info(f"{self.task_id} completed successfully.")
-                return {"Training": serialize(description)}
-
-            self.defer(
-                timeout=self.execution_timeout,
-                trigger=SagemakerTrigger(
-                    poke_interval=self.check_interval,
-                    end_time=end_time,
-                    aws_conn_id=self.aws_conn_id,
-                    job_name=self.config["TrainingJobName"],
-                    job_type="Training",
-                    response_key="TrainingJobStatus",
-                ),
-                method_name="execute_complete",
-            )
-
-        # for bypassing mypy missing return error
-        return None  # pragma: no cover
-
-    def execute_complete(self, context: Context, event: dict[str, Any]) -> dict[str, Any]:  # type: ignore[override]
-        """
-        Callback for when the trigger fires - returns immediately.
-        Relies on trigger to throw an exception, otherwise it assumes execution was
-        successful.
-        """
-        if event and event["status"] == "success":
-            self.log.info("%s completed successfully.", self.task_id)
-            return {"Training": serialize(event["message"])}
-        if event and event["status"] == "error":
-            raise AirflowException(event["message"])
-        raise AirflowException("No event received in trigger callback")
+        super().__init__(deferrable=True, **kwargs)
```

Comment:
shall we pass args too?

Reply:
this is not needed, the upstream operator has no positional args. It only has named/keyword args.